packages/wonder-blocks-data/src/util/__tests__/response-cache.test.js

```diff
@@ -436,105 +459,371 @@ describe("../response-cache.js", () => {
+                    // Act
+                    const result = cache.getEntry(fakeHandler, "options");
+
+                    // Assert
+                    expect(result).toBeNull();
+                });
+
+                it("should return the cached entry if in the hydration cache", () => {
+                    // Arrange
+                    const hydrationCache = new MemoryCache();
+                    jest.spyOn(hydrationCache, "retrieve").mockReturnValue({
+                        data: "data!",
+                    });
+                    const cache = new ResponseCache(hydrationCache);
+                    const fakeHandler: IRequestHandler<string, string> = {
+                        getKey: () => "MY_KEY",
+                        type: "MY_HANDLER",
+                        fulfillRequest: jest.fn(),
+                        hydrate: true,
+                    };
+
+                    // Act
+                    const result = cache.getEntry(fakeHandler, "options");
+
+                    // Assert
+                    expect(result).toStrictEqual({data: "data!"});
+                });
+            });
+
+            describe("handler does not want hyrdation", () => {
+                it("should return undefined", () => {
+                    // Arrange
+                    const hydrationCache = new MemoryCache();
+                    jest.spyOn(hydrationCache, "retrieve").mockReturnValue({
+                        data: "data!",
+                    });
+                    const cache = new ResponseCache(hydrationCache);
+                    const fakeHandler: IRequestHandler<string, string> = {
+                        getKey: () => "MY_KEY",
+                        type: "MY_HANDLER",
+                        fulfillRequest: jest.fn(),
+                        hydrate: false,
+                    };
+
+                    // Act
+                    const result = cache.getEntry(fakeHandler, "options");
+
+                    // Assert
+                    expect(result).toBeUndefined();
+                });
+            });
         });
 
-        it("should return the cached entry if in the hydration cache", () => {
-            // Arrange
-            const internalCache = new MemoryCache();
-            jest.spyOn(internalCache, "retrieve").mockReturnValue({
-                data: "data!",
+        describe("when server-side", () => {
+            beforeEach(() => {
+                jest.spyOn(Server, "isServerSide").mockReturnValue(true);
             });
-            const cache = new ResponseCache(internalCache);
-            const fakeHandler: IRequestHandler<string, string> = {
-                getKey: () => "MY_KEY",
-                type: "MY_HANDLER",
-                fulfillRequest: jest.fn(),
-                hydrate: true,
-            };
 
-            // Act
-            const result = cache.getEntry(fakeHandler, "options");
+            describe("handler wants hydration", () => {
```

Comment:
style nit: This describe block appears to be the same for client-side and server-side.  You could use a `describeEach` to de-dupe these.  This would also make it more explicit that this behaviour is the same between the two environments.

Reply:
I kept them separate because they are semantically different and could easily diverge. I don't feel strongly about it, but I have a mild preference for the separation and the clarity it has in telling a story (plus matching the pattern of other tests in the file).